File: FragmentActivityDemo/app/src/main/java/com/example/fragmentactivitydemo/MainActivity.java
package com.example.fragmentactivitydemo;

import android.support.v4.app.FragmentManager;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;

public class MainActivity extends AppCompatActivity {
      static final String TAG_CHILD_FRAGMENT = "child_fragment";
      static final String TAG_INNER_FRAGMENT = "inner_fragment";

    private ChildFragment mChildFragment;
    private InnerFragment mInnerFragment;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        FragmentManager fm = getSupportFragmentManager();
       // mChildFragment=new ChildFragment();

        //fm.beginTransaction().replace(R.id.fragment_contaner,mChildFragment,TAG_CHILD_FRAGMENT).commit();
        mChildFragment = (ChildFragment) fm.findFragmentByTag(TAG_CHILD_FRAGMENT);
        if(mChildFragment==null){
            mChildFragment=new ChildFragment();
            fm.beginTransaction().add(R.id.fragment_contaner,mChildFragment,TAG_CHILD_FRAGMENT).commit();
            mInnerFragment=new InnerFragment();
            fm.beginTransaction().add(R.id.fragment_contaner,mInnerFragment,TAG_INNER_FRAGMENT).commit();
        }

    }
}

File: BLE/app/src/main/java/com/example/ble/MainActivity.kt
package com.example.ble

import android.support.v7.app.AppCompatActivity
import android.os.Bundle

class MainActivity : AppCompatActivity() {
    companion object {
        var fragment = BlankFragment()
        var fragment2 = BlankFragment2()
        var current = 1
    }

    var fragmentstr = ""

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        /*if(savedInstanceState!=null){
           when{
               savedInstanceState.getString("fragment").contentEquals("1")->{
                   fragment.arguments!!.putBundle("savedInstanceState",savedInstanceState)
                   supportFragmentManager.beginTransaction().replace(R.id.framelayoutContaner,  fragment).commit()

               }
               savedInstanceState.getString("fragment").contentEquals("2")->{
                   fragment2.arguments!!.putBundle("savedInstanceState",savedInstanceState)
                   supportFragmentManager.beginTransaction().replace(R.id.framelayoutContaner, fragment2).commit()
               }
           }

        }*/
        when (current) {
            1 -> {
                supportFragmentManager.beginTransaction().replace(R.id.framelayoutContaner, fragment).commit()
            }
            2 -> {
                supportFragmentManager.beginTransaction().replace(R.id.framelayoutContaner, fragment2).commit()
            }
        }
    }

    override fun onSaveInstanceState(savedInstanceState: Bundle) {
        super.onSaveInstanceState(savedInstanceState)
        if (!savedInstanceState.containsKey("fragment")) {
            savedInstanceState.putString("fragment", "1")
        }
    }

    override fun onRestoreInstanceState(savedInstanceState: Bundle) {
        super.onRestoreInstanceState(savedInstanceState)
        fragmentstr = savedInstanceState.getString("fragment")
    }
}
